### DISCUSSION TITLE
Use the second USB serial?

### DISCUSSION THREAD

**Discussion by foxt** [micropython/micropython#12552] [RP2040 / Pico] [answered]
Hello, under CircuitPython, you can have two USB serials running on the 2040. I was wondering if there was a way to do this on MicroPython? It's very useful as you can have the first serial console be debug logging & REPL where have the second one be 100% dedicated to data transfer.

https://docs.circuitpython.org/en/latest/shared-bindings/usb_cdc/index.html

**Comment by shariltumin** [comment]
You can have 2 UART in MicroPython

```
MPY: soft reboot
MicroPython v1.20.0 on 2023-04-26; Raspberry Pi Pico W with RP2040
Type "help()" for more information.
>>> from machine import UART
>>> uart0 = UART(0)
>>> uart1 = UART(1)
>>> print(uart0)
UART(0, baudrate=115200, bits=8, parity=None, stop=1, tx=0, rx=1, txbuf=256, rxbuf=256, timeout=0, timeout_char=1, invert=None)
>>> print(uart1)
UART(1, baudrate=115200, bits=8, parity=None, stop=1, tx=4, rx=5, txbuf=256, rxbuf=256, timeout=0, timeout_char=1, invert=None)
>>>
```

**Reply by robert-hh** [reply]
The request was for two serial channels by the single USB connector, not for UART interfaces.

**Reply by shariltumin** [reply]
Sorry, my bad.

**Comment by peterhinch** [comment, marked as answer]
@foxt That is a very neat feature. Sadly I don't think MP supports it.

**Reply by foxt** [reply]
Ah. That's unfortunate. Appreciate your response anyway.

**Comment by jimmo** [comment]
@foxt This is supported in MicroPython on other boards (e.g. stm32 boards such as the pyboard).

Much more expanded support for USB device classes (e.g. midi, hid, etc) as well as defining additional CDC interfaces will be supported on rp2/Pico soon. See #9497
